### PR TITLE
e2e: enhance volumegroupsnapshot test

### PR DESCRIPTION
**Pull request by Madhu-1**
update the e2e for volumegroupsnapshot to create application pods from clone PVC and delete the pods once we are able to create the pods.

### CODE REVIEW

**Comment by Madhu-1** [comment]
@nixpanic i added few more steps to create pod and delete pod for the cloned volumes (part of volumegroupsnapshot)

File: e2e/volumegroupsnapshot_base.go [hunk 1]
```diff
@@ -199,6 +199,70 @@ func (v *volumeGroupSnapshotterBase) CreatePVCClones(
...
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("pod-%d", i),
+				Namespace: p.Namespace,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:    "container",
+						Image:   "quay.io/centos/centos:latest",
+						Command: []string{"/bin/sleep", "999999"},
```
Comment (nixpanic) [inline e2e/volumegroupsnapshot_base.go:215]: use `sleep infinity` instead?

**Review by nixpanic — APPROVED** [review]
LGTM, mentioned just a nit

File: e2e/volumegroupsnapshot_base.go (hunk 1 above)
Comment (Madhu-1) [reply e2e/volumegroupsnapshot_base.go:215]: i think should be fine for now, i just copied it from other test file, planning for refractoring will take care of it in next PR.

**Comment by nixpanic** [comment]
Tested with RBD VolumeGroupSnapshot in [PR#4502](https://github.com/ceph/ceph-csi/pull/4502#issuecomment-2450551899).

**Review by iPraveenParihar — APPROVED** [review]
LGTM

**Comment by Madhu-1** [comment]
@mergifyio queue

**Pull request merged**